### DISCUSSION TITLE
Make ArchR project from Signac-analysed scATAC-seq

### DISCUSSION THREAD

**Discussion by AnnaAMonaco** [GreenleafLab/ArchR#1740] [Questions / Documentation] [answered]
Dear Dev Team,

I was wondering if there is a way to make an ArchR project from a scATAC-seq already analysed in Signac (normalised, dim reduced, clustered, etc), so basically a Seurat object. 

I have a dataset that I already annotated cell types to and I already performed some downstream pseudo-bulk analysis using these cluster annotations to aggregate info. I would like to use ArchR's co-accessibility and gene2peak linkage functions, but for me it is essential that I keep the same clustering and annotations as in my Seurat object. 

I looked a lot online and in this discussion section, but I could not find a way to create an ArchR project from an already processed dataset, but since addGeneIntegrationMatrix( ) can read Seurat objects for scRNA-seq, I was wondering if there would be a way to read Seurat objects for scATAC-seq too. Or alternatively, any other way to keep previously performed clustering and annotations.

Thanks in advance!

Anna

**Comment by rcorces** [comment, marked as answer]
This is not possible in the way you would like. Your best bet is to create and ArchRProject from scratch using the same data that you analyzed in Signac. Then import the dimensionality reduction or embedding into the ArchRProject and any annotations you want into `cellColData`.
The importing of the dimensionality reduction is not documented anywhere and I dont have code to illustrate how it would be done because I dont use Signac. Importing an embedding is comparatively easier but that may not be sufficient for your purposes. I'm pasting an excerpt from an in-progress update to the manual that shows how to import an embedding.

```
## Importing an embedding from external software

If you have analyzed data using another package, you can always import your embedding coordinates into ArchR. To do this, you need to create the various attributes that ArchR expects of an embedding and then add this to the `embeddings` slot of your `ArchRProject`.

The key component of an embedding in ArchR is a `DataFrame` that contains the cell names and the embedding coordinates of each cell. In the below example, we will randomly generate arbitrary UMAP coordinates for all of the cells in `projHeme5` but in your own application you would need to provide UMAP coordinates ordered in the same order as returned by `getCellNames()`.

#simulate mock UMAP data by random sampling
umap1 <- (sample(x = -100000:100000, size = length(getCellNames(projHeme2)), replace = FALSE))/10000
umap2 <- (sample(x = -100000:100000, size = length(getCellNames(projHeme2)), replace = FALSE))/10000

#add embedding to projHeme2
umap_df <- DataFrame(row.names = getCellNames(projHeme2), "custom#UMAP1" = umap1, "custom#UMAP2" =  umap2, check.names = FALSE)
projHeme2@embeddings$customUMAP <- SimpleList(df = umap_df, params = list())


With this, we would be able to use this embedding by name as with any other embedding in our `ArchRProject`. It is important to note that the embedding coordinate column names (i.e `"custom#UMAP1"` and `"custom#UMAP2"` in the example above) must contain the `#` in the name as this is used in string splitting operations in `plotEmbedding()` and other functions.

getEmbedding(ArchRProj = projHeme5, embedding = "customUMAP")

```

**Reply by AnnaAMonaco** [reply]
Actually, I think this should work for me! I'll definitely give it a try, thank you. My main concern is that the annotated clusters remain the same and this seems to do the job :)

Cheers,
Anna

**Reply by drosop** [reply]
Hi [AnnaAMonaco](https://github.com/AnnaAMonaco),

I am trying to do a similar analysis. I have signac preprocessed object. Did you find a solution? Thanks,
